src/firewall-config

```diff
@@ -1614,7 +1618,15 @@ class FirewallConfig(object):
                         buttons=(("gtk-ok", 0),("gtk-quit", 1))) == 1:
             self.onQuit()
 
+    def connection_failed(self, msg):
+        if self._dialog("<b>"+_("Error")+"</b>",
+                        message_type=Gtk.MessageType.ERROR, msg=msg,
+                        buttons=[("gtk-quit", 1)]) == 1:
+            self.onQuit()
+
     def connection_changed(self):
+        if self.connection_timer:
+                GLib.source_remove(self.connection_timer)
```

Comment:
Do we need to set self.connection_timer to None so we don't re-trigger source_remove() ?